### DISCUSSION TITLE
Wyze Cam V3 RTSP "Stream Not Found"

### DISCUSSION THREAD

**Discussion by krishnatrivedi** [gtxaspec/wz_mini_hacks#394] [Q&A] [answered]
Greetings,

I'm trying to setup RTSP with Shinobi. I've configured the settings in wz_mini.conf as such:

> #####VIDEO STREAM#####      
RTSP_LOGIN="admin"           
RTSP_PASSWORD="abc"         
RTSP_PORT="8554"
                       
> RTSP_HI_RES_ENABLED="true"
RTSP_HI_RES_ENABLE_AUDIO="true"
RTSP_HI_RES_FPS="20"
RTSP_HI_RES_MAX_BITRATE="2048"
RTSP_HI_RES_TARGET_BITRATE="1024"
RTSP_HI_RES_ENC_PARAMETER="2"
           
> RTSP_LOW_RES_ENABLED="false"
RTSP_LOW_RES_ENABLE_AUDIO="false"    
RTSP_LOW_RES_FPS=""    
RTSP_LOW_RES_MAX_BITRATE=""     
RTSP_LOW_RES_TARGET_BITRATE=""     
RTSP_LOW_RES_ENC_PARAMETER=""        
                            
> RTSP_AUTH_DISABLE="false" 

I've confirmed with Netcat that I can connect to the device via port 8554. I edited the config file directly on the device via SSH.

However, when I try to connect in Shinobi or VLC, I get this error:

> [rtsp @ 0x558b01d9e280] method DESCRIBE failed: 404 Stream Not Found rtsp://admin:abc@192.168.1.167:8554/video1_unicast: Server returned 404 Not Found 

Am I doing something wrong?

Thanks in advance for any help.

Regards,
KT

**Comment by virmaior** [comment, marked as answer]
1. what's the firmware version on the V3?
2. If you open the same stream on VLC, what are the results?

**Reply by krishnatrivedi** [reply]
Thanks for the reply. Answers below and additional detail provided below that.

1. The firmware version is:  4.36.10.2864
2. VLC:
    
In the UI:
> Your input can't be opened:
   VLC is unable to open the MRL 'rtsp://admin:abc@192.168.1.167:8554/video1_unicast'. Check the log for details.
  
In the VLC log file:  
> satip error: Failed to setup RTSP session






Additional information from the device itself (via SSH) and Shinobi logs:

In the v4l2rtspserver.log file:

> handleCmd_SETUP:SETUP rtsp://192.168.1.167:8554/unicast/track1 RTSP/1.0
Transport: RTP/AVP/TCP;unicast;interleaved=0-1
CSeq: 4
User-Agent: Lavf58.20.100
Authorization: Digest username="admin", realm="LIVE555 Streaming Media", nonce="2b5a06cc3734021cfb8e6dc5c006e056", uri="rtsp://192.168.1.167:8554/unicast/track1", response="d378dce1d3af7f7aac87cdc5e592e13e"

I also tried the URl listed above but with the same error in Shinobi:

URI listed in documentation:
> [rtsp @ 0x55f4a2d23380] method DESCRIBE failed: 404 Stream Not Found rtsp://admin:abc@192.168.1.167:8554/video1_unicast: Server returned 404 Not Found 

URI in the v4l2rtspserver.log file:
> [rtsp @ 0x555acba53380] method DESCRIBE failed: 404 Stream Not Found rtsp://admin:abc@192.168.1.167:8554/unicast/track1: Server returned 404 Not Found 

In the wz_init.log file:

> #####S15v4l2rtspserver#####
Enable video ch 0
Enable audio ch 0
rtsp low_res disabled
delay RTSP for iCamera
#####S09coredump#####
Killing dumpload
Setting kernel core pattern
: IP Address not acquired yet, try again in 5 seconds
S12remoteaccessory: IP Address not acquired yet, try again in 5 seconds
Starting RTSP server
Set imp variables via helper
#####S16rtmp#####
K01network: IP Address not acquired yet, try again in 5 seconds
#####S09dnsmasq#####
#####S12remoteaccessory#####
Remote accessory support disabled
No custom script configured in wz_mini.conf
Store dmesg logs
Deleting logs older than 5 boots...
sh: /opt/wz_mini/log: out of range
sh: /opt/wz_mini/log/wz_init.log: out of range
sh: /opt/wz_mini/log/v4l2rtspserver.log: out of range
sh: /opt/wz_mini/log/dmesg.log: out of range

**Reply by virmaior** [reply]
4.36.10.2864 doesn't work with wz_mini at least not with the rtsp functionality. This is right on the main page:

> Works on ANY firmware release up to 4.36.9.139 (DO NOT UPGRADE BEYOND THIS FW IF YOU WANT RTSP or any cmd FEATURES!)

**Reply by krishnatrivedi** [reply]
Ah ok. I'll downgrade the firmware.

**Reply by krishnatrivedi** [reply]
Greetings,

OK, so I've tried on another camera. 

Firmware is 4.36.9.139. It's a Wyze Cam V3.

Config in /media/mmc/wz_mini/wz_mini.conf:

>#####VIDEO STREAM#####    
RTSP_LOGIN="admin"                   
RTSP_PASSWORD="abc12345"    
RTSP_PORT="8554"                
                                   
>RTSP_HI_RES_ENABLED="true"           
RTSP_HI_RES_ENABLE_AUDIO="true"
RTSP_HI_RES_FPS=""        
RTSP_HI_RES_MAX_BITRATE=""           
RTSP_HI_RES_TARGET_BITRATE=""
RTSP_HI_RES_ENC_PARAMETER=""    
                                   
>RTSP_LOW_RES_ENABLED="true"          
RTSP_LOW_RES_ENABLE_AUDIO="true"
RTSP_LOW_RES_FPS=""       
RTSP_LOW_RES_MAX_BITRATE=""          
RTSP_LOW_RES_TARGET_BITRATE=""
RTSP_LOW_RES_ENC_PARAMETER=""   
                                   
>RTSP_AUTH_DISABLE="false"       

VLC log (UI):  

>Your input can't be opened:
VLC is unable to open the MRL 'rtsp://admin:abc12345@192.168.1.169:8554/video1_unicast'. Check the log for details.     

VLC log (text output):

>satip error: Failed to setup RTSP session

Shinboi log:

>[rtsp @ 0x55c1d9339380] method DESCRIBE failed: 404 Stream Not Found rtsp://admin:abc12345@192.168.1.169:8554/video1_unicast: Server returned 404 Not Found 

Am I missing something in the config file?

Thanks in advance for any tips.

Regards,
KT

**Reply by clee604** [reply]
I'm not sure if they are mandatory fields but you are missing values for max_bitrate, target_bitrate and en_parameter

**Reply by krishnatrivedi** [reply]
Thanks @clee604 and @virmaior I finally got it working. 

It was, in fact, the missing parameters. I put in the values I tried the first time with the updated firmware and now it works. 

Most appreciated.

For my edification, are there plans to enable RSTP  with firmware version 4.36.10.2864?

Regards,
KT
